platform/core/src/services/DisplaySetService/DisplaySetService.js

```diff
@@ -245,4 +250,45 @@ export default class DisplaySetService {
     }
     return allDisplaySets;
   }
+
+
+  filterDisplaySetInstances(displaySetInstanceUID, filteredInstances) {
+    const displaySet = this.getDisplaySetByUID(displaySetInstanceUID);
+    let changed = false;
+
+    if (displaySet) {
+      const instances = displaySet.images || displaySet.others || [];
+
+      if (instances.length === filteredInstances.length) return; //shortcut
+      for (let index = instances.length - 1; index >= 0; index--) {
```

Comment:
Is it not possible to use instances.filter here? Perhaps I am missing something?

Reply:
instances.filter may work too if it is supported on both "images" and "others".  I think they are a simple arrays so it should but doing a loop will work just fine too.